app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt

```diff
@@ -90,114 +224,169 @@ class AutoCompleteApi @Inject constructor(
+                    }
+                } + entries.filterIsInstance<VisitedPage>()
+        }
+            .map { entry ->
+                when (entry) {
+                    is VisitedPage -> {
+                        AutoCompleteHistorySuggestion(
+                            phrase = entry.url.toStringDropScheme(),
+                            title = entry.title,
+                            url = entry.url.toString(),
+                            isAllowedInTopHits = isAllowedInTopHits(entry),
+                        )
+                    }
+                    is VisitedSERP -> {
+                        AutoCompleteHistorySearchSuggestion(
+                            phrase = entry.query,
+                            isAllowedInTopHits = isAllowedInTopHits(entry),
+                        )
+                    }
+                }.let { suggestion ->
+                    RankedSuggestion(suggestion, autoCompleteScorer.score(entry.title, entry.url, entry.visits.size, query))
+                }
+            }.filter { it.score > 0 }
             .toList()
     }
 
     private fun scoreTitle(
-        rankedBookmark: RankedBookmark,
+        rankedBookmark: RankedSuggestion<AutoCompleteBookmarkSuggestion>,
         query: String,
-    ): RankedBookmark {
-        if (rankedBookmark.savedSite.title.startsWith(query, ignoreCase = true)) {
-            rankedBookmark.score += 200
-        } else if (rankedBookmark.savedSite.title.contains(" $query", ignoreCase = true)) {
-            rankedBookmark.score += 100
+    ): RankedSuggestion<AutoCompleteBookmarkSuggestion> {
+        return if (rankedBookmark.suggestion.title.startsWith(query, ignoreCase = true)) {
+            rankedBookmark.copy(score = rankedBookmark.score + 200)
+        } else if (rankedBookmark.suggestion.title.contains(" $query", ignoreCase = true)) {
+            rankedBookmark.copy(score = rankedBookmark.score + 100)
+        } else {
+            rankedBookmark
         }
-
-        return rankedBookmark
     }
 
     private fun scoreTokens(
-        rankedBookmark: RankedBookmark,
+        rankedBookmark: RankedSuggestion<AutoCompleteBookmarkSuggestion>,
         query: String,
-    ): RankedBookmark {
-        val savedSite = rankedBookmark.savedSite
-        val domain = savedSite.url.toUri().baseHost
+    ): RankedSuggestion<AutoCompleteBookmarkSuggestion> {
+        val suggestion = rankedBookmark.suggestion
+        val domain = suggestion.url.toUri().baseHost
         val tokens = query.split(" ")
+        var toReturn = rankedBookmark
         if (tokens.size > 1) {
             tokens.forEach { token ->
-                if (!savedSite.title.startsWith(token, ignoreCase = true) &&
-                    !savedSite.title.contains(" $token", ignoreCase = true) &&
+                if (!suggestion.title.startsWith(token, ignoreCase = true) &&
+                    !suggestion.title.contains(" $token", ignoreCase = true) &&
```

Comment:
Found a tiny bug here, not new but it's an easy fix:
For bookmarks that are not urls (e.g: bookmarklets), they don't have domain. `domain?.startsWith(token, ignoreCase = true) == false` won't be true because domain is null, and it will show as suggestion always.

`(domain == null || !domain.startsWith(token, ignoreCase = true))`